### ISSUE TITLE
Get selection from an internal command such as "commandline get-selection"

### ISSUE THREAD

**Issue by blindFS** [nushell/reedline#829]
I'd like to add keybinding features like vim-surround to vi normal mode.
As far as I know, I can't get selected text or copied text (not the system copy/cut) in pure nushell script.
It would be even better if `commandline edit` support more actions such as those already existed in `keybindings list --edits`.

## References

https://github.com/tpope/vim-surround

**Comment by fdncred** [comment]
I'm guessing that we'd support such a functionality. You can see all the commandline commands now with `help commandline` in nushell. maybe a `commandline set-selection` and `commandline get-selection` to follow the current commands?

**Comment by blindFS** [comment]
Just checked the source code on the main branch. Only get-cursor and set-cursor are implemented. I'd like to have 2-int (start, end) version of them.

**Comment by fdncred** [comment]
Right, I was suggesting that you write some new commands to support the functionality you're describing.

**Comment by blindFS** [comment]
Hi @fdncred , I'd like to discuss the design choice here. Why we use a ReplState instead of just a Reedline editor in struct EngineState? Seems that the set-selection function (or any other feature that requires reedline editor involved) is impossible to implement in trait Command?

**Comment by fdncred** [comment]
I'm not sure. I don't understand reedline's internals enough to help answer.
